x/nodes/keeper/valStateChanges_test.go

```diff
@@ -350,6 +352,129 @@ func TestValidatorStateChange_EditAndValidateStakeValidatorAfterNonCustodialUpgr
 	}
 }
 
+func runStake(ctx sdk.Context,
+	k Keeper, msg types.MsgStake, signer crypto.PublicKey) sdk.Error {
+	validator := types.NewValidator(
+		sdk.Address(msg.PublicKey.Address()),
+		msg.PublicKey,
+		msg.Chains,
+		msg.ServiceUrl,
+		sdk.ZeroInt(),
+		msg.Output)
+	err := k.ValidateValidatorStaking(
+		ctx, validator, msg.Value, sdk.Address(signer.Address()))
+	if err != nil {
+		return err
+	}
+	return k.StakeValidator(ctx, validator, msg.Value, signer)
+}
+
+func Test_OutputAddressChange(t *testing.T) {
+	ctx, _, k := createTestInput(t, true)
+
+	// Enable EditStake
+	codec.UpgradeHeight = -1
+
+	stakeAmount := sdk.NewCoin(k.StakeDenom(ctx), sdk.NewInt(k.MinimumStake(ctx)))
+	outputAccount := getRandomPubKey()
+	outputAddress := sdk.Address(outputAccount.Address())
+	msgTemplate := types.MsgStake{
+		Chains:     []string{"0021", "0040"},
+		ServiceUrl: "https://www.pokt.network:443",
+		Value:      stakeAmount.Amount,
+	}
+
+	// Create and fund accounts
+	operatorAcc1 := getRandomPubKey()
+	operatorAcc2 := getRandomPubKey()
+	operatorAcc3 := getRandomPubKey()
+	operatorAddr1 := sdk.Address(operatorAcc1.Address())
+	operatorAddr2 := sdk.Address(operatorAcc2.Address())
+	operatorAddr3 := sdk.Address(operatorAcc3.Address())
+	assert.Nil(t, fundAccount(ctx, k, operatorAddr1, stakeAmount))
+	assert.Nil(t, fundAccount(ctx, k, operatorAddr2, stakeAmount))
+	assert.Nil(t, fundAccount(ctx, k, outputAddress, stakeAmount))
+
+	// Stake two nodes before NCUST
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = outputAddress
+	assert.Nil(t, runStake(ctx, k, msgTemplate, operatorAcc1))
+	msgTemplate.PublicKey = operatorAcc2
+	msgTemplate.Output = outputAddress
+	assert.Nil(t, runStake(ctx, k, msgTemplate, operatorAcc2))
+
+	// Verify staked nodes having nil output address
+	validatorCur, found := k.GetValidator(ctx, operatorAddr1)
+	assert.True(t, found)
+	assert.Nil(t, validatorCur.OutputAddress)
+	validatorCur, found = k.GetValidator(ctx, operatorAddr2)
+	assert.True(t, found)
+	assert.Nil(t, validatorCur.OutputAddress)
+
+	// Enable NCUST
+	codec.UpgradeFeatureMap[codec.NonCustodialUpdateKey] = -1
+
+	// Set an output address on Addr1 after NCUST
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = outputAddress
+	assert.Nil(t, runStake(ctx, k, msgTemplate, operatorAcc1))
+	validatorCur, found = k.GetValidator(ctx, operatorAddr1)
+	assert.True(t, found)
+	assert.True(t, validatorCur.OutputAddress.Equals(outputAddress))
```

Comment:
Use `assert.Equal` directly

Reply:
Oh I didn't know `assert.Equal` is that smart.  Good to know.